autoload/clap/cache.vim

```diff
@@ -0,0 +1,54 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: Cache API for clap.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+let s:path_separator = has('win32') ? '\' : '/'
+let s:clap_cache_dir = get(g:, 'clap_cache_dir', '')
+
+function! clap#cache#folder() abort
+  if empty(s:clap_cache_dir)
+    if has('nvim')
+      let user_cache = stdpath('cache')
+    elseif exists('$XDG_CACHE_HOME')
+      let user_cache = $XDG_CACHE_HOME
+    else
+      let user_cache = $HOME . s:path_separator . '.cache'
+    endif
+    let s:clap_cache_dir = user_cache . s:path_separator . 'clap'
+  endif
+
+  if !isdirectory(s:clap_cache_dir)
+    call mkdir(s:clap_cache_dir, 'p')
+  endif
+
+  return s:clap_cache_dir
+endf
+
+function! clap#cache#file(cache_name) abort
+  return clap#cache#folder() . s:path_separator . a:cache_name
+endfunction
+
+function! clap#cache#read(cache_name) abort
```

Comment:
I'm unsure if this is neccessary to provide `clap#cache#read` and `clap#cache#write` as we already provided the cache file path via `clap#cache#file()`.

Reply:
It's okay for me, readfile() and writefile() is easy enough. The developer only need to compose their cache filename. If you prefer to remove it, I can remove it.

Reply:
Done!